src/Middleware/WebSockets/src/HandshakeHelpers.cs

```diff
@@ -72,5 +75,186 @@ public static string CreateResponseKey(string requestKey)
+                    // "client_max_window_bits" extension parameter, the corresponding
+                    // extension negotiation response to the offer MUST NOT include the
+                    // "client_max_window_bits" extension parameter.
+                    builder.Append("; ").Append(WebSocketDeflateConstants.ClientMaxWindowBits).Append('=')
+                        .Append(parsedOptions.ClientMaxWindowBits.ToString(CultureInfo.InvariantCulture));
+                }
+                else if (value.StartsWith(WebSocketDeflateConstants.ServerMaxWindowBits))
+                {
+                    // https://datatracker.ietf.org/doc/html/rfc7692#section-7
+                    // MUST decline if:
+                    // The negotiation offer contains multiple extension parameters with
+                    // the same name.
+                    if (hasServerMaxWindowBits)
+                    {
+                        return false;
+                    }
+
+                    hasServerMaxWindowBits = true;
+                    if (!ParseWindowBits(value, WebSocketDeflateConstants.ServerMaxWindowBits, out var parsedServerMaxWindowBits))
+                    {
+                        return false;
+                    }
+
+                    // 8 is a valid value according to the spec, but our zlib implementation does not support it
+                    if (parsedServerMaxWindowBits == 8)
+                    {
+                        return false;
+                    }
+
+                    // https://tools.ietf.org/html/rfc7692#section-7.1.2.1
+                    // A server accepts an extension negotiation offer with this parameter
+                    // by including the "server_max_window_bits" extension parameter in the
+                    // extension negotiation response to send back to the client with the
+                    // same or smaller value as the offer.
+                    parsedOptions.ServerMaxWindowBits = Math.Min(parsedServerMaxWindowBits ?? 15, serverMaxWindowBits);
+                }
+
+                static bool ParseWindowBits(ReadOnlySpan<char> value, string propertyName, out int? parsedValue)
+                {
+                    var startIndex = value.IndexOf('=');
+
+                    // parameters can be sent without a value by the client, we'll use the values set by the app developer or the default of 15
+                    if (startIndex < 0)
+                    {
+                        parsedValue = null;
+                        return true;
+                    }
+
+                    value = value[(startIndex + 1)..].TrimEnd();
+
+                    // https://datatracker.ietf.org/doc/html/rfc7692#section-5.2
+                    // check for value in quotes and pull the value out without the quotes
+                    if (value[0] == '"' && value.EndsWith("\"".AsSpan()))
+                    {
+                        value = value[1..^1];
+                    }
```

Comment:
Does this handle the error case of a single quote `foo="`?

Reply:
Without a number? Will add a test for that